In [4]:
# Import Libraries
import pandas as pd
import numpy as np


In [ ]:
# Read data
df = pd.read_csv('P2M3_gracia_valerine_data_raw.csv', sep=",", encoding='Latin-1')
df

ORDERNUMBER  QUANTITYORDERED  PRICEEACH  ORDERLINENUMBER    SALES  \
0           10107               30      95.70                2  2871.00   
1           10121               34      81.35                5  2765.90   
2           10134               41      94.74                2  3884.34   
3           10145               45      83.26                6  3746.70   
4           10159               49     100.00               14  5205.27   
...           ...              ...        ...              ...      ...   
2818        10350               20     100.00               15  2244.40   
2819        10373               29     100.00                1  3978.51   
2820        10386               43     100.00                4  5417.57   
2821        10397               34      62.24                1  2116.16   
2822        10414               47      65.52                9  3079.44   

            ORDERDATE    STATUS  QTR_ID  MONTH_ID  YEAR_ID  ...  \
0      2/24/2003 0:00   Shipped       1         2     2003  ...   
1       5/7/2003 0:00   Shipped       2         5     2003  ...   
2       7/1/2003 0:00   Shipped       3         7     2003  ...   
3      8/25/2003 0:00   Shipped       3         8     2003  ...   
4     10/10/2003 0:00   Shipped       4        10     2003  ...   
...               ...       ...     ...       ...      ...  ...   
2818   12/2/2004 0:00   Shipped       4        12     2004  ...   
2819   1/31/2005 0:00   Shipped       1         1     2005  ...   
2820    3/1/2005 0:00  Resolved       1         3     2005  ...   
2821   3/28/2005 0:00   Shipped       1         3     2005  ...   
2822    5/6/2005 0:00   On Hold       2         5     2005  ...   

                       ADDRESSLINE1  ADDRESSLINE2           CITY STATE  \
0           897 Long Airport Avenue           NaN            NYC    NY   
1                59 rue de l'Abbaye           NaN          Reims   NaN   
2     27 rue du Colonel Pierre Avia           NaN          Paris   NaN   
3                78934 Hillside Dr.           NaN       Pasadena    CA   
4                   7734 Strong St.           NaN  San Francisco    CA   
...                             ...           ...            ...   ...   
2818             C/ Moralzarzal, 86           NaN         Madrid   NaN   
2819                    Torikatu 38           NaN           Oulu   NaN   
2820             C/ Moralzarzal, 86           NaN         Madrid   NaN   
2821          1 rue Alsace-Lorraine           NaN       Toulouse   NaN   
2822             8616 Spinnaker Dr.           NaN         Boston    MA   

     POSTALCODE  COUNTRY TERRITORY CONTACTLASTNAME CONTACTFIRSTNAME DEALSIZE  
0         10022      USA       NaN              Yu             Kwai    Small  
1         51100   France      EMEA         Henriot             Paul    Small  
2         75508   France      EMEA        Da Cunha           Daniel   Medium  
3         90003      USA       NaN           Young            Julie   Medium  
4           NaN      USA       NaN           Brown            Julie   Medium  
...         ...      ...       ...             ...              ...      ...  
2818      28034    Spain      EMEA          Freyre            Diego    Small  
2819      90110  Finland      EMEA       Koskitalo           Pirkko   Medium  
2820      28034    Spain      EMEA          Freyre            Diego   Medium  
2821      31000   France      EMEA          Roulet          Annette    Small  
2822      51003      USA       NaN         Yoshido             Juri   Medium  

[2823 rows x 25 columns]

In [ ]:
df['SALES'].max()

14082.8

In [ ]:
df.isnull().sum()

ORDERNUMBER            0
QUANTITYORDERED        0
PRICEEACH              0
ORDERLINENUMBER        0
SALES                  0
ORDERDATE              0
STATUS                 0
QTR_ID                 0
MONTH_ID               0
YEAR_ID                0
PRODUCTLINE            0
MSRP                   0
PRODUCTCODE            0
CUSTOMERNAME           0
PHONE                  0
ADDRESSLINE1           0
ADDRESSLINE2        2521
CITY                   0
STATE               1486
POSTALCODE            76
COUNTRY                0
TERRITORY           1074
CONTACTLASTNAME        0
CONTACTFIRSTNAME       0
DEALSIZE               0
dtype: int64

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2823 entries, 0 to 2822
Data columns (total 25 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ORDERNUMBER       2823 non-null   int64  
 1   QUANTITYORDERED   2823 non-null   int64  
 2   PRICEEACH         2823 non-null   float64
 3   ORDERLINENUMBER   2823 non-null   int64  
 4   SALES             2823 non-null   float64
 5   ORDERDATE         2823 non-null   object 
 6   STATUS            2823 non-null   object 
 7   QTR_ID            2823 non-null   int64  
 8   MONTH_ID          2823 non-null   int64  
 9   YEAR_ID           2823 non-null   int64  
 10  PRODUCTLINE       2823 non-null   object 
 11  MSRP              2823 non-null   int64  
 12  PRODUCTCODE       2823 non-null   object 
 13  CUSTOMERNAME      2823 non-null   object 
 14  PHONE             2823 non-null   object 
 15  ADDRESSLINE1      2823 non-null   object 
 16  ADDRESSLINE2      302 non-null    object 


To be fixed columns:

5   ORDERDATE         2823 non-null   object  -> to datetime
16  ADDRESSLINE2      302 non-null    object  -> to drop
18  STATE             1337 non-null   object  -> to drop
19  POSTALCODE        2747 non-null   object  -> to drop
21  TERRITORY         1749 non-null   object  -> to drop 

Will be dropped because these columns have repititive information that is invaluable and will not disrupt the flow. 

In [ ]:
   # Load the data


    # Clean by dropping duplicates, formatting the data, and dropping unused/empty columns data
df.drop_duplicates(inplace=True)
df.columns = df.columns.str.lower().str.replace(' ', '_').str.strip()
df.drop(columns=['addressline2', 'state', 'territory', 'postalcode'], inplace=True)

    # Turning the cleaned data into new csv file
df.to_csv('P2M3_gracia_valerine_data_clean.csv', index=False)

In [5]:
ds = pd.read_csv('P2M3_gracia_valerine_data_clean.csv')
ds

ordernumber  quantityordered  priceeach  orderlinenumber    sales  \
0           10107               30      95.70                2  2871.00   
1           10121               34      81.35                5  2765.90   
2           10134               41      94.74                2  3884.34   
3           10145               45      83.26                6  3746.70   
4           10159               49     100.00               14  5205.27   
...           ...              ...        ...              ...      ...   
2818        10350               20     100.00               15  2244.40   
2819        10373               29     100.00                1  3978.51   
2820        10386               43     100.00                4  5417.57   
2821        10397               34      62.24                1  2116.16   
2822        10414               47      65.52                9  3079.44   

            orderdate    status  qtr_id  month_id  year_id  ... msrp  \
0      2/24/2003 0:00   Shipped       1         2     2003  ...   95   
1       5/7/2003 0:00   Shipped       2         5     2003  ...   95   
2       7/1/2003 0:00   Shipped       3         7     2003  ...   95   
3      8/25/2003 0:00   Shipped       3         8     2003  ...   95   
4     10/10/2003 0:00   Shipped       4        10     2003  ...   95   
...               ...       ...     ...       ...      ...  ...  ...   
2818   12/2/2004 0:00   Shipped       4        12     2004  ...   54   
2819   1/31/2005 0:00   Shipped       1         1     2005  ...   54   
2820    3/1/2005 0:00  Resolved       1         3     2005  ...   54   
2821   3/28/2005 0:00   Shipped       1         3     2005  ...   54   
2822    5/6/2005 0:00   On Hold       2         5     2005  ...   54   

      productcode              customername             phone  \
0        S10_1678         Land of Toys Inc.        2125557818   
1        S10_1678        Reims Collectables        26.47.1555   
2        S10_1678           Lyon Souveniers  +33 1 46 62 7555   
3        S10_1678         Toys4GrownUps.com        6265557265   
4        S10_1678  Corporate Gift Ideas Co.        6505551386   
...           ...                       ...               ...   
2818     S72_3212     Euro Shopping Channel    (91) 555 94 44   
2819     S72_3212   Oulu Toy Supplies, Inc.        981-443655   
2820     S72_3212     Euro Shopping Channel    (91) 555 94 44   
2821     S72_3212              Alpha Cognac        61.77.6555   
2822     S72_3212         Gifts4AllAges.com        6175559555   

                       addressline1           city  country contactlastname  \
0           897 Long Airport Avenue            NYC      USA              Yu   
1                59 rue de l'Abbaye          Reims   France         Henriot   
2     27 rue du Colonel Pierre Avia          Paris   France        Da Cunha   
3                78934 Hillside Dr.       Pasadena      USA           Young   
4                   7734 Strong St.  San Francisco      USA           Brown   
...                             ...            ...      ...             ...   
2818             C/ Moralzarzal, 86         Madrid    Spain          Freyre   
2819                    Torikatu 38           Oulu  Finland       Koskitalo   
2820             C/ Moralzarzal, 86         Madrid    Spain          Freyre   
2821          1 rue Alsace-Lorraine       Toulouse   France          Roulet   
2822             8616 Spinnaker Dr.         Boston      USA         Yoshido   

     contactfirstname dealsize  
0                Kwai    Small  
1                Paul    Small  
2              Daniel   Medium  
3               Julie   Medium  
4               Julie   Medium  
...               ...      ...  
2818            Diego    Small  
2819           Pirkko   Medium  
2820            Diego   Medium  
2821          Annette    Small  
2822             Juri   Medium  

[2823 rows x 21 columns]

In [ ]:
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2823 entries, 0 to 2822
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ordernumber       2823 non-null   int64  
 1   quantityordered   2823 non-null   int64  
 2   priceeach         2823 non-null   float64
 3   orderlinenumber   2823 non-null   int64  
 4   sales             2823 non-null   float64
 5   orderdate         2823 non-null   object 
 6   status            2823 non-null   object 
 7   qtr_id            2823 non-null   int64  
 8   month_id          2823 non-null   int64  
 9   year_id           2823 non-null   int64  
 10  productline       2823 non-null   object 
 11  msrp              2823 non-null   int64  
 12  productcode       2823 non-null   object 
 13  customername      2823 non-null   object 
 14  phone             2823 non-null   object 
 15  addressline1      2823 non-null   object 
 16  city              2823 non-null   object 


In [7]:
ds['orderdate'].max()

'9/9/2004 0:00'

Notes: 

OBJECTIVE:

This project is built to analyze sales performance, customer, product, order, profitability, revenue trends, and business strategy development. By assessing overall sales performance, understanding customer demographics and purchasing behavior, evaluating product performance and popularity, and analyzing order detail, trends, and revenue, we could develop effective business strategy based on data insights to improve sales and customer satisfaction.

